### PR TITLE
Enhancement/1097 mark optimization

### PR DESCRIPTION
**Pull request by RidhamShah**

### CODE REVIEW

File: backend/packages/Upgrade/src/api/services/SegmentService.ts [hunk 1]
```diff
@@ -62,6 +62,7 @@ export class SegmentService {
 
   public async getSegmentByIds(ids: string[]): Promise<Segment[]> {
     return this.cacheService.wrapFunction(CACHE_PREFIX.SEGMENT_KEY_PREFIX, ids, async () => {
+      console.log('ids', ids);
```
Comment (ppratikcr7) [inline backend/packages/Upgrade/src/api/services/SegmentService.ts:65]: remove this console log

File: backend/packages/Upgrade/src/api/services/SegmentService.ts [hunk 2]
```diff
@@ -74,7 +75,7 @@ export class SegmentService {
       const sortedData = ids.map((id) => {
         return result.find((data) => data.id === id);
       });
-
+      console.log('sortedData', sortedData);
```
Comment (ppratikcr7) [inline backend/packages/Upgrade/src/api/services/SegmentService.ts:78]: remove this console log

File: backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts [hunk 3]
```diff
@@ -292,10 +263,10 @@ export class ExperimentAssignmentService {
         relations: ['user'],
       });
 
-      if (experimentDecisionPoint.length && experimentId) {
-        const selectedExperimentDP = experimentDecisionPoint.filter((dp) => dp.experiment.id === experimentId);
+      if (experimentId && experiments.length > 0) {
+        const selectedExperimentDP = dpExperiments.filter((dp) => dp.experiment.id === experimentId);
```
Comment (VivekFitkariwala) [inline backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts:267]: Use find instead of filter

**Review by VivekFitkariwala — CHANGES_REQUESTED** [review]
Some small requested. PR looks good.

File: backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts (hunk 3 above)
Comment (RidhamShah) [reply backend/packages/Upgrade/src/api/services/ExperimentAssignmentService.ts]: code updated

**Review by VivekFitkariwala — APPROVED** [review]

**Pull request merged**